src/ShellCommand/ShellCommandRunner.php

```diff
@@ -307,18 +313,10 @@ private function _uploadToS3($localFilePath, $targetUrl)
 
     $bucket   = $urlParts['host'];
     $path     = preg_replace('/^\//', '', $urlParts['path']);
-    $headers  = array(
-      'fileUpload' => $localFilePath,
-      'headers'    => array(),
-    );
 
     // Upload!
-    $s3       = new AmazonS3($creds);
-    $response = $s3->create_object($bucket, $path, $headers);
-    if (!$response->isOK())
-    {
-      throw new Exception("Upload to S3 {$targetUrl} failed: " . print_r($response->body, true) . PHP_EOL . print_r($response->header, true));
-    }
+    $s3     = S3Client::factory($creds);
+    $result = $s3->upload($bucket . 'sdfjb', $path, $localFilePath);  // throws on error
```

Comment:
and it looks like it's documented that all errors are returned thru ["general error codes" http://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html]